cmake/configure/configure_1_cuda.cmake

```diff
@@ -39,13 +41,62 @@ MACRO(FEATURE_CUDA_FIND_EXTERNAL var)
 
   IF(CUDA_FOUND)
 
-    SET(CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE FALSE)
+    SET(${var} TRUE)
 
-    SET(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} -std=c++11)
+    IF("${CUDA_NVCC_FLAGS}" MATCHES "-arch")
 
-    # FIXME: CUDA compiler NVCC doesn't support C++14.
+      # Compute Capability specified explicitly.
+      # Now parse:
 
-    SET(${var} TRUE)
+      IF("${CUDA_NVCC_FLAGS}" MATCHES "-arch[ ]*sm_([0-9]*)")
+        SET(CUDA_COMPUTE_CAPABILITY "${CMAKE_MATCH_1}")
+      ELSEIF("${CUDA_NVCC_FLAGS}" MATCHES "-arch=sm_([0-9]*)")
+        SET(CUDA_COMPUTE_CAPABILITY "${CMAKE_MATCH_1}")
+      ELSE()
+        STRING(REGEX MATCH "(-arch[ ]*[^ ]*)" match "${CUDA_NVCC_FLAGS}")
+        MESSAGE(STATUS "Missformed Compute Capability specified.")
+        SET(CUDA_ADDITIONAL_ERROR_STRING
+          ${CUDA_ADDITIONAL_ERROR_STRING}
+          "A missformed Compute Capability was passed in CUDA_NVCC_FLAGS: ${match}\n"
+          "deal.II requires at least Compute Capability 3.5\n"
+          "which is used as default is nothing is specified."
+          )
+        SET(${var} FALSE)
+      ENDIF()
+
+
+      IF("${CUDA_COMPUTE_CAPABILITY}" LESS "35")
+        MESSAGE(STATUS "Too low CUDA Compute Capability specified -- deal.II requires at least 3.5 ")
+        SET(CUDA_ADDITIONAL_ERROR_STRING
+          ${CUDA_ADDITIONAL_ERROR_STRING}
+          "Too low CUDA Compute Capability specified: ${CUDA_COMPUTE_CAPABILITY}\n"
+          "deal.II requires at least Compute Capability 3.5\n"
+          "which is used as default is nothing is specified."
+          )
+        SET(${var} FALSE)
+      ENDIF()
+    ENDIF()
+
+    # Configuration was successful
+    IF(${var})
```

Comment:
At some point we should factor the stuff inside the `IF(${var})` - `ENDIF()` into a macro ```
```
MACRO(FEATURE_CUDA_CONFIGURE_EXTERNAL)
  ...
ENDMACRO()
```